### ISSUE TITLE
git history lost

### ISSUE THREAD

**Issue by rofl0r** [ttcdt/mp-5.x#2]
hi, thanks for keeping this project alive. unfortunately you abandoned the entire git history (i happen to have [an old fork](https://github.com/rofl0r/mp-5.x) that has the entire history up to 5.2.10). this seems unfortunate, as having the commit history allows one to read the code like a book and often find out the exact reasoning for a change on per-line granularity.
also the AUTHORS file refers one to look at the git log which is now pointless. may i ask what was the reasoning behind this move ?

**Comment by ttcdt** [comment]
Hi. It's a personal preference of the original programmer, Angel Ortega.
When he left the project he asked us to keep rebasing all changes and
only push finished releases into Github, and we agreed (this repo is
just a mirror of a private one).

Regarding the AUTHORS file, you're right that it doesn't make any sense
as it is now. We'll fix it.

Regards,
Theo

4/11/20 4:08, rofl0r:

**Comment by rofl0r** [comment]
thanks for your response. btw, you're not "watching" the https://github.com/ttcdt/mpsl repo, so you won't be notified of new issues filed there (like the one i opened).

meanwhile i managed to update the git history of my fork up to 11/2019 by merging what was available on another fork. this allowed me to figure out how the build error in #3 was introduced.

**Comment by ttcdt** [comment]
Certainly, watch on the mpsl repo was not enabled (don't know why).
Thanks for bringing this to our attention.

Do you have a patch to fix #3 issue in your fork?

6/11/20 13:14, rofl0r:

**Comment by rofl0r** [comment]
> Do you have a patch to fix #3 issue in your fork?

no, my idea was to create a "gtk3to2" wrapper which would automatically make gtk+3 apps work with a gtk+2 install (by having a fake pkgconfig file that returns gtk+2.0 CFLAGS plus -include gtk3to2.h below), however this turned out to be a lot harder than i hoped.

```c
#ifndef GTK3TO2_H
#define GTK3TO2_H

//#include "gtk3to2_keysyms.h"

#include <gtk-2.0/gtk/gtkversion.h>

#undef GTK_MAJOR_VERSION
#define GTK_MAJOR_VERSION 3

#define GDK_SCROLL_SMOOTH -1
#define GDK_SMOOTH_SCROLL_MASK 0

struct _GdkRGBA
{
  /*g*/double red;
  double green;
  double blue;
  double alpha;
};
typedef struct _GdkRGBA GdkRGBA;

#define gtk3to2_colval(COLVAL) (COLVAL*65535.0)
#define gtk3to2_rgba2color(COL) &(GdkColor) {\
	.red = gtk3to2_colval((COL)->red), \
	.green = gtk3to2_colval((COL)->green), \
	.blue = gtk3to2_colval((COL)->blue), \
	} /* fixme: pixel uint32 */
#define gdk_cairo_set_source_rgba(CR, COL) \
  gdk_cairo_set_source_color(CR, gtk3to2_rgba2color(COL))

#define gtk_scrollbar_new(O, N) \
(O == GTK_ORIENTATION_VERTICAL ? gtk_vscrollbar_new(N) : gtk_hscrollbar_new(N))

//#define GdkRGBA GdkColor
#define gtk_widget_get_allocated_width(AREA) (AREA)->allocation.width
#define gtk_widget_get_allocated_height(AREA) (AREA)->allocation.height

#endif
```
i've ranted about gtk+3 here in my blog: http://sabotage-linux.neocities.org/blog/8

**Comment by ttcdt** [comment]
Fine. I'm not a GTK expert, but Tiny (another collaborator in our team)
is. I know he's pretty busy these days but I'm sure he's reading this
and get back to us with a solution. I'll also ask Angel but he is not
much into development these days.

I also find GTK cumbersome and its unexpected changes annoying, I'm more
a Qt dude.

Theo
